helper.php

```diff
@@ -134,5 +134,104 @@ public function hasUserAcknowledged($page, $user)
 
         return $acktime ? (int)$acktime : false;
     }
+
+    /**
+     * Save user's acknowledgement for a given page
+     *
+     * @param string $page
+     * @param string $user
+     * @return bool
+     */
+    public function saveAcknowledgement($page, $user)
+    {
+        $sqlite = $this->getDB();
+        if (!$sqlite) return false;
+
+        $sql = "REPLACE INTO acks (page, user, ack) VALUES (?,?, strftime('%s','now'))";
+
+        $result = $sqlite->query($sql, $page, $user);
+        $sqlite->res_close($result);
+        return true;
+
+    }
+
+    /**
+     * Fetch all assignments for a given user, with additional page information.
+     *
+     * @param string $user
+     * @return array|bool
+     */
+    public function getUserAssignments($user)
+    {
+        $sqlite = $this->getDB();
+        if (!$sqlite) return false;
+
+        global $USERINFO;
+        $groups = $USERINFO['grps'];
+
+        $sql = "SELECT * FROM assignments A
+                JOIN pages B
+                ON A.page = B.page
+                WHERE A.assignee LIKE ? OR A.assignee LIKE ?";
+
+        $userWildcard = '%' . $user . '%';
+        $groupWildcard = '%@' . $groups . '%';
+
+        $result = $sqlite->query($sql, $userWildcard, $groupWildcard);
+        $assignments = $sqlite->res2arr($result);
+        $sqlite->res_close($result);
+
+        return $assignments;
+    }
+
+    /**
+     * Compare user's assignments with their past acknowledgements and return only pages
+     * without or with outdated acknowledgements
+     *
+     * @param string $user
+     * @param array $assignments
+     * @return array|bool
+     */
+    public function filterAcknowledged($user, $assignments)
```

Comment:
This one could be combined with the method above, I believe?

Reply:
I removed the method completely because the filtering is now done in the SQL query.